datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/internal/BaseWritableBufferImpl.java

```diff
@@ -387,9 +387,13 @@ final void putNativeOrderedShort(final long offsetBytes, final short value) {
     unsafe.putShort(getUnsafeObject(), getCumulativeOffset(offsetBytes), value);
   }
 
-  //OTHER
-  @Override
-  public final Object getArray() {
+  //OTHER WRITE METHODS
+
+  /**
+   * Returns the primitive backing array, otherwise null.
+   * @return the primitive backing array, otherwise null.
+   */
```

Comment:
Does checkValid result in an exception if the memory is invalid?  Do we follow any convention on declaring that a method may throw an exception in these cases?

Reply:
Yes, that is its purpose. As for the declaration part, that is a very good catch.  I will have to go through nearly all the methods add a declaration to that effect.  -- But I will do that in the next phase.   --Thanks, this was very helpful!